src/components/ui/modal.tsx

```diff
@@ -17,6 +17,18 @@ export default function Modal({
   children: React.ReactNode
   props?: any
 }) {
+  useEffect(() => {
+    const handleKeyDown = (event: KeyboardEvent) => {
+      if (event.key === 'Escape' && !hideCloseButton) {
+        close(event)
```

Comment:
adding `event` to `close` to match the calling signature of `onClick={close}` is v smart :)